plutarch-trace/src/Plutarch/TraceSig.hsig

```diff
@@ -0,0 +1,18 @@
+signature Plutarch.TraceSig (ptrace', ptrace, ptraceIfTrue, ptraceIfFalse) where
+
+import Prelude ()
+import Plutarch.Prelude (Term, (:-->))
+import Plutarch.String (PString)
+import Plutarch.Bool (PBool)
+
+-- | A strict version of 'ptrace'.
+ptrace' :: Term s (PString :--> a :--> a)
```

Comment:
I just realised that you also have `ptrace'` here. When would you actually use this?

Reply:
Whenever anyone wants a strict version, and *possibly* slightly more efficient version - maybe. Similar to `pif'`, `pand'`, and the like. `ptrace` delays its second argument (Haskell level argument) before application, and then finally forces before returning. This was to ensure that the log within `ptrace` happens *before* the potential logs within its second argument.

If `ptrace'` has no chance of being more efficient in a case like `ptrace' # "foo" # ()` - where delaying the second argument is useless - it makes sense to remove it. But I'm not sure if that's the case.

Reply:
No, it *is* more efficient in some cases, but we don't care about efficiency in the first place when traces are enabled, no? Traces are going to be completely removed in production after all.

Reply:
https://github.com/Plutonomicon/plutarch/pull/39/commits/98b8e5f47f5bea7ac9af8513ee76734461a51742